### PR TITLE
fix: preserve SQL NULL in parquet export

### LINKED ISSUE #686: [BUG] Parquet export converts SQL NULL cells to empty strings
## Summary
Exporting query results to Parquet loses SQL `NULL` values. After a round-trip through `--parquet --output` (or `.mode parquet` + `.dump`), a `NULL` cell comes back as an empty string.

This breaks one of sqly's recent guarantees for data-analysis workflows: `NULL` and `""` should stay distinguishable in machine-readable output.

## Reproduction
```bash
tmp=$(mktemp -d)
./sqly --parquet --output "$tmp/nulls.parquet" \
  --sql "SELECT CAST(NULL AS TEXT) AS id, 'A' AS name UNION ALL SELECT '1' AS id, 'B' AS name"
./sqly --json-typed --sql 'SELECT * FROM "nulls"' "$tmp/nulls.parquet"
```

## Actual result
```json
[
  {"id":"","name":"A"},
  {"id":1,"name":"B"}
]
```

The first row's `id` was `NULL`, but it round-trips as `""`.

## Expected result
The first row should remain `NULL` after Parquet export and re-import:
```json
[
  {"id":null,"name":"A"},
  {"id":1,"name":"B"}
]
```

## Likely cause
`infrastructure/filesql/DumpTableToParquet()` stages the result into a temporary SQLite DB using `GenerateInsertStatement(table.Name(), record)`, which only sees `[]string` cell values and has no access to `table.IsNull(row, col)`. That collapses `NULL` and empty string before the Parquet writer runs.

## Tests to add
1. Add a regression test in `infrastructure/filesql/parquet_test.go` that builds a `model.Table` with a null bitmap (`SetNulls`) and asserts that a Parquet round-trip preserves `NULL` vs empty string.
2. Add a higher-level shell test that exports a query result containing `NULL` with `--parquet --output`, re-imports it, and verifies `--json-typed` still emits `null`.
3. Extend the existing Parquet round-trip test to assert cell values, not just row count and column names.

### PR DESCRIPTION
**Pull request by nao1215**
## Summary

A SQL `NULL` exported to Parquet round-tripped back as an empty string, so `NULL` and `""` became indistinguishable in machine-readable output. For the issue's example, the first row's `NULL` id came back as `""`.

Two layers caused the loss:
1. sqly's export staging built every cell as a quoted string literal, collapsing a `NULL` into `''`.
2. filesql wrote a null cell to Parquet as `""` and reloaded it as `""`.

## Changes

- The Parquet staging insert now emits SQL `NULL` for cells the result marks as NULL (`parquetInsertStatement` consults `table.IsNull`).
- Bumps `github.com/nao1215/filesql` to v0.15.0, which preserves the null through the Parquet write and reload (nullable Arrow column on write, per-cell null mask on the streaming read).

A `NULL` now round-trips as `null` while an empty string stays `""`. For the issue's example, `--json-typed` re-emits `{"id":null,...}` then `{"id":1,...}`.

## Tests

Unit: `parquet_test.go` builds a table with a null bitmap (`SetNulls`) and asserts NULL versus empty string survive the round-trip.
E2E: `spec/parquet_export_spec.sh` exports a NULL and verifies `--json-typed` re-emits `null` and not `""`.

Depends on filesql v0.15.0 (released). Companion: nao1215/filesql#136.

Closes #686

### CODE REVIEW

File: infrastructure/filesql/parquet_test.go [hunk 1]
```diff
@@ -140,6 +140,72 @@ func TestDumpTableToParquet_PreservesNumericLookingText(t *testing.T) {
...
+// column's values as sql.NullString, so a test can tell a SQL NULL apart from an
+// empty string after a round-trip.
+func reimportColumn(t *testing.T, parquetPath, tableName, column string) []sql.NullString {
+	t.Helper()
+	db, err := libfilesql.OpenContext(context.Background(), parquetPath)
+	if err != nil {
+		t.Fatalf("reimport parquet: %v", err)
+	}
+	defer func() { _ = db.Close() }()
+	rows, err := db.QueryContext(context.Background(), "SELECT "+column+" FROM "+tableName)
```
[the remark this answers is not in the document]
Comment (nao1215) [reply infrastructure/filesql/parquet_test.go:153]: False positive in test code, consistent with the existing reimportRowCount and reimportStringColumn helpers in this file. reimportColumn builds the query from a constant column and table name the test passes, not untrusted input, and the CI golangci-lint job is green. Leaving it unannotated keeps the test helpers consistent.
